### PR TITLE
Moved animations to utilities and spinner functionality

### PR DESCRIPTION
**Pull request by jay24rajput**
Moved animations.py from plugins to utilities folder. Added functionality for spinner to directly be invoked by jarvis in JarvisAPI.

Closes Issue #596

### CODE REVIEW

**Comment by jay24rajput** [comment]
The travis is failing because of import statements from `plugins.animation` which has been now shifted to `utilities.animation`

**Comment by pnhofmann** [comment]
> The travis is failing because of import statements from plugins.animation which has been now shifted to utilities.animation

That is strange... But probably nothing to worry about. And actually the test is failing because of PEP8:

```
 flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/

jarviscli/CmdInterpreter.py:127:16: E225 missing whitespace around operator
jarviscli/CmdInterpreter.py:127:43: E231 missing whitespace after ','
jarviscli/CmdInterpreter.py:133:45: E231 missing whitespace after ','
jarviscli/CmdInterpreter.py:192:27: E231 missing whitespace after ','
jarviscli/CmdInterpreter.py:197:45: E231 missing whitespace after ','
jarviscli/CmdInterpreter.py:200:26: E231 missing whitespace after ','
jarviscli/CmdInterpreter.py:206:25: E231 missing whitespace after ','
The command "flake8 --select E,W --max-line-length=140 --ignore E722,W503,W504,E128 jarviscli/" exited with 1.
```

But thanks! Just like I imagined!

If you like, found two bugs with SpinningThread:

1. 
  * Run ``speedtest``
  * Immediately cancel with Str-C
2.
  * Disconnect from internet
  * Run ``football`` 

Both cases will result in infinity spinning cursor. In case you would like to do something more ( ;) ) finding a way to make sure that SpinningThread stops in all cases would be a task. In paticular:

* If Jarvis exits
* If plugin runs in expected Network error (https://github.com/sukeesh/Jarvis/blob/master/jarviscli/CmdInterpreter.py#L82)
* If plugin runs in unexpected exception (https://github.com/sukeesh/Jarvis/blob/master/jarviscli/CmdInterpreter.py#L186)

**Comment by jay24rajput** [comment]
Okay! I will look into these details and issue the PR :smile:

File: jarviscli/plugins/voice.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def disable_sound(jarvis, s):
 
 
 @plugin('say')
-def say(jarvis, s):
```
Comment (pnhofmann) [inline jarviscli/plugins/voice.py]: This change I don't understand. And breaks say-command.

But otherwise great work!

Comment (jay24rajput) [reply jarviscli/plugins/voice.py]: Sorry! Might have changed it mistakenly changed it while breaking the code :sweat_smile: I will fix it with a commit!

**Comment by pnhofmann** [comment]
Then all what's left for me is to say Thanks and merge ;).

**Pull request merged**